src/index.ts

```diff
@@ -342,11 +345,57 @@ task("starknet-deploy", "Deploys Starknet contracts which have been compiled.")
                     ],
                     [artifactsPath]
                 );
-
-                return processExecuted(executed);
+                if(args.wait){
+                    const execResult = processExecuted(executed);
+                    
+                    if(execResult == 0)
+                        tx_hashes.push(executed.stdout.toString().match(".*Transaction hash: (?<tx_hash>\\w*).*").groups["tx_hash"]);
+                    
+                    return execResult;
+                } 
+                else
+                    return processExecuted(executed);
             });
         }
 
+        if(args.wait){      //  If the "wait" flag was passed as an argument, check the previously stored transaction hashes for their statuses
+            let done = false;
+            var hash_done:boolean [] = [];
+            for(var i = 0; i<tx_hashes.length;i++){
+                hash_done.push(false);
+            }
+            while(!done){   // Run loop until all transactions are PENDING or ACCEPTED
```

Comment:
This action of the starknet-verify task is quiet big. Some of its sections should be extracted to separate functions.